File: Niveau 1/7 – Conditions avancées, opérateurs booléens/06 - Casernes de pompiers.py
nbPaire = int(input())
for i in range(nbPaire):
    minX1 = int(input())
    maxX1 = int(input())
    minY1 = int(input())
    maxY1 = int(input())
    
    minX2 = int(input())
    maxX2 = int(input())
    minY2 = int(input())
    maxY2 = int(input())
    if ((maxX1 < minX2) or maxX1 == minX2) or ((maxX2 < minX1) or maxX2 == minX1):
       print("NON")
    elif ((maxY1 < minY2) or maxY1 == minY2) or ((maxY2 < minY1) or maxY2 == minY1):
       print("NON")
    else:
       print("OUI")

File: Niveau 1/7 – Conditions avancées, opérateurs booléens/06 - Casernes de pompiers.cpp
#include <iostream>
using namespace std;
int main()
{
  int nbPaire;
  cin >> nbPaire;
  for (int i = 0; i < nbPaire; i++)
  {
    int minX1, maxX1, minY1, maxY1;
    int minX2, maxX2, minY2, maxY2;
    cin >> minX1 >> maxX1 >> minY1 >> maxY1;
    cin >> minX2 >> maxX2 >> minY2 >> maxY2;
    if (((maxX1 < minX2) || maxX1 == minX2) || ((maxX2 < minX1) || maxX2 == minX1))
    {
      cout << "NON" << endl;
    }
    else if (((maxY1 < minY2) || maxY1 == minY2) || ((maxY2 < minY1) || maxY2 == minY1))
    {
      cout << "NON" << endl;
    }
    else
    {
      cout << "OUI" << endl;
    }
  }
}